bundles/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/handler/SpotifyBridgeHandler.java

```diff
@@ -340,22 +354,32 @@ private boolean pollStatus() {
         synchronized (pollSynchronization) {
             try {
                 onAccessTokenResponse(getAccessTokenResponse());
-                // Collect devices and populate selection with available devices.
-                final List<Device> ld = devicesCache.getValue();
-                final List<Device> devices = ld == null ? Collections.emptyList() : ld;
-                spotifyDynamicStateDescriptionProvider.setDevices(devicesChannelUID, devices);
                 // Collect currently playing context.
                 final CurrentlyPlayingContext pc = playingContextCache.getValue();
-                final CurrentlyPlayingContext playingContext = pc == null ? EMPTY_CURRENTLYPLAYINGCONTEXT : pc;
-                final List<Playlist> lp = playlistCache.getValue();
-                final List<Playlist> playlists = lp == null ? Collections.emptyList() : lp;
-                updateStatus(ThingStatus.ONLINE);
+                // If Spotify returned a 204. Meaning everything is ok, but we got no data.
+                // Happens when no song is playing. And we know no device was active
+                // No need to continue because no new information will be available.
+                final boolean hasPlayData = pc != null && pc.getDevice() != null;
+                final CurrentlyPlayingContext playingContext = pc == null ? EMPTY_CURRENTLY_PLAYING_CONTEXT : pc;
 
-                handleCommand.setLists(devices, playlists);
-                updatePlayerInfo(playingContext, playlists);
-                spotifyDynamicStateDescriptionProvider.setPlayLists(playlistsChannelUID, playlists);
+                // Collect devices and populate selection with available devices.
+                if (hasPlayData || hasAnyDeviceStatusUnknown()) {
+                    final List<Device> ld = devicesCache.getValue();
+                    final List<Device> devices = ld == null ? Collections.emptyList() : ld;
+                    spotifyDynamicStateDescriptionProvider.setDevices(devicesChannelUID, devices);
+                    handleCommand.setDevices(devices);
+                    updateDevicesStatus(devices, playingContext.isPlaying());
```

Comment:
The only statements about which I doubt whether it should be nulled out when receiving a 204 from spotify are this one and maybe line 379. WDYT?

Reply:
The 204 happen after some time of inactivity. So If openHAB is running at that point the status is already everything is inactive. When openHAB is started and Spotify returns 204 than the status is set because the UNKNOWN state is true. So I would think no other action is needed.